model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/add_extra_diffusion_for_normal_wind_tendency_approaching_cfl.py

```diff
@@ -57,19 +57,28 @@ def _add_extra_diffusion_for_normal_wind_tendency_approaching_cfl(
         0.0,
     )
     difcoef = where(
-        abs(w_con_e) > astype(cfl_w_limit * ddqz_z_full_e, wpfloat),
+        (levelmask | levelmask(Koff[1]))
+        & (abs(w_con_e) > astype(cfl_w_limit * ddqz_z_full_e, wpfloat)),
```

Comment:
Can we get rid of the condition here though?
As soon as `difcoef` is inlined, it will never be computed as long as the condition in the `ddt_vn_apc_wp` computation is false, correct?

Reply:
Have you checked if this calculation gets still inlined without this change in the if statement?

Reply:
I checked and indeed this is not necessary